workflow/rules/common.smk

```diff
@@ -1026,115 +1040,318 @@ def get_trimmed_fastqs(wc):
+            "ANN['CLIN_SIG']",
+            "ANN['gnomADg_AF']",
+            "ANN['EXON']",
+            "ANN['REVEL']",
+            # variants only, split-call-tables.py will select the column with the
+            # highest score and will put it in the same place
+            "ANN['SpliceAI_pred_DS_AG']",
+            "ANN['SpliceAI_pred_DS_AL']",
+            "ANN['SpliceAI_pred_DS_DG']",
+            "ANN['SpliceAI_pred_DS_DL']",
+            # variants only
+            "ANN['am_pathogenicity']",
+        ]
+    )
+    # COLLAPSED variants columns
+    sort_order.extend(
+        [
+            # variants & fusions
+            "CHROM",
+            "POS",
+            # variants only
+            "REF",
+            "ALT",
+            "ANN['Protein_position'].raw",
+            "ANN['Amino_acids']",
+            "ANN['CANONICAL']",
+            "ANN['MANE_PLUS_CLINICAL']",
+            # fusions only
+            "INFO['GENE_NAME']",
+            "INFO['GENE_ID']",
+        ]
+    )
+    sort_order.extend(
+        # fusions only
+        get_and_sort_multi_entry_fields("INFO['EXON_NUMBER'][0]")
+    )
+    sort_order.extend(
+        # MAIN column for fusions, COLLAPSED column for variants
+        get_and_sort_multi_entry_fields("INFO['PROB_")
+    )
+    sort_order.extend(
+        # MAIN column for fusions & variants
+        # gets moved ahead of consequence column for variants
+        get_and_sort_multi_entry_fields("FORMAT['AF']")
+    )
+    sort_order.extend(
+        # MAIN column for fusions, COLLAPSED column for variants
+        get_and_sort_multi_entry_fields("FORMAT['DP']")
+    )
+    # COLLAPSED columns
+    sort_order.extend(
+        # fusions only, join key in join_fusion_partner.py
+        get_and_sort_multi_entry_fields("INFO['MATEID'][0]")
+    )
+    sort_order.extend(
+        # variants & fusions
+        get_and_sort_multi_entry_fields("FORMAT['SROBS']")
+    )
+    sort_order.extend(
+        # variants & fusions
+        get_and_sort_multi_entry_fields("FORMAT['SAOBS']")
+    )
+    sort_order.extend(
+        # variants & fusions
+        get_and_sort_multi_entry_fields("FORMAT['OBS']")
+    )
+    sort_order.extend(
+        [
+            # only needed for fusions, as a join key; dropped in
+            # split-call-tables.py
+            "ID",
+            # only needed for indexing of variants in split-call-tables.py
+            "ANN['Gene']",
+            # only needed for variants in oncoprint.py
+            "ANN['HGVSg']",
+            # only needed for variants in datavzrd_variants_calls
+            "ANN['Feature']",
+            "INFO['END']",
+        ]
+    )
+    # sort columns, keeping only those in sort_order
+    sorted_columns_dict = {k: columns_dict[k] for k in sort_order if k in columns_dict}
+    join_items = ", ".join
+    return {
+        "expr": join_items(sorted_columns_dict.keys()),
+        "header": join_items(sorted_columns_dict.values()),
+    }
```

Comment:
1. Why the sorting?
2. The code is quite redundant. Maybe use a loop over the different args that get_and_sort_multi_entry_fields is called with?

Reply:
1. I added some comments here: https://github.com/snakemake-workflows/dna-seq-varlociraptor/pull/297/commits/b14d8cfa1411fedb9fc5f3b717b225023341fba5
2. I reduced a bit of redundancy here: https://github.com/snakemake-workflows/dna-seq-varlociraptor/pull/297/commits/735624117f409f99abea8ef731de1d25a928c362